pkg/curatedpackages/bundle.go

```diff
@@ -29,18 +39,20 @@ type BundleRegistry interface {
 type BundleReader struct {
 	kubeConfig    string
 	source        BundleSource
-	kubectl       KubectlRunner
 	bundleManager Manager
 	registry      BundleRegistry
+	restClient    rest.Interface
```

Comment:
Shouldn't we be able to specify the behavior we need and the client implementation is then irrelevant to this consumer?

Reply:
Just had a good chat over chime, which I'll summarize here:

+ We agreed that the code would be improved if this used a more task-centered interface of our own definition.
+ We also agreed that such an interface wouldn't be entirely in the scope of this PR, and is probably best added in a follow-up.

Furthermore, part of the reason for the 2nd point above is because we don't have a good grasp of what our interface would look like. Using the k8s client-go interface is still somewhat exploratory, so the above gives us freedom for now, though it does also leave exposed some rough edges that if not handled carefully could bite us later.

... hopefully I can get to the follow on soon.